### PR TITLE
Add localized text support 

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #103 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is, and anything that describes the before/after state of the project._

The scope of this project is to introduce a new `lang` property at the element scope which will, on load, trigger the placement of predefined copy in the requested language. 

Current languages supported are `en` and `es`. 

Predefined strings and their localized versions were moved from being internal to the element to an external js file. While the current scope only supports `en` and `es`, additional languages can be supported with little additional effort. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: test/auro-input.test.js [hunk 1]
```diff
@@ -225,6 +225,16 @@ describe('auro-input', () => {
     });
   });
 
+  describe('handles i18n', () => {
+    it('credit-card translation', async () => {
+      const el = await fixture(html`
+        <auro-input type="credit-card" required lang="es" id="input01"></auro-input>
+      `);
+
+      expect(el.shadowRoot.querySelector('#input01')).to.have.attribute('lang', 'es');
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply test/auro-input.test.js:234]: That is a great question... I will see what I can do.

File: test/auro-input.test.js [hunk 2]
```diff
@@ -228,13 +228,24 @@ describe('auro-input', () => {
...
+        <auro-input type="credit-card" required id="input01"></auro-input>
+      `);
+
+      const eli18n = await fixture(html`
         <auro-input type="credit-card" required lang="es" id="input01"></auro-input>
       `);
 
-      expect(el.shadowRoot.querySelector('#input01')).to.have.attribute('lang', 'es');
+      const eli18nContent = eli18n.shadowRoot.querySelectorAll('[name="helptext"]')[0].innerHTML;
+      const content = el.shadowRoot.querySelectorAll('[name="helptext"]')[0].innerHTML;
```
Comment (geoffrich) [inline test/auro-input.test.js:239]: nit: just use `querySelector` here if we're getting the first element
```suggestion
      const eli18nContent = eli18n.shadowRoot.querySelector('[name="helptext"]').innerHTML;
      const content = el.shadowRoot.querySelector('[name="helptext"]').innerHTML;
```

File: demo/demo.md [hunk 3]
```diff
@@ -140,6 +140,16 @@ Use the `borderless` attribute to remove the bottom border. This attribute is in
   ```
 </auro-accordion>
 
+## Localization support
+
+The `auro-input` element supports the localization of all content managed within the scope of the element. This DOES NOT include any custom content places in the `slot` element.
```
Comment (geoffrich) [inline demo/demo.md:145]:
```suggestion
The `auro-input` element supports the localization of all content managed within the scope of the element. This DOES NOT include any custom content placed in the `slot` element.
```

File: src/base-input.js [hunk 4]
```diff
@@ -416,18 +417,18 @@ export default class BaseInput extends LitElement {
   }
 
   /**
-   * Determines default help text string from this.localContent.
+   * Determines default help text string from this.strings object.
```
Comment (geoffrich) [inline src/base-input.js:420]:
```suggestion
   * Determines default help text string.
```

File: src/i18n.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Function to support the selected of a string in the set lang.
+ * @param {string} lang - Requested lang for content return.
+ * @param {string} requestedString - String requested in context.
+ * @private
+ * @returns {string} Value of string request.
+ */
+export default function i18n(lang, requestedString) {
+
```
Comment (geoffrich) [inline src/i18n.js:67]:
```suggestion
```

File: demo/index.html [hunk 6]
```diff
@@ -16,6 +16,14 @@
...
     <main></main>
 
+    <!-- move this to an external example -->
+    <!-- <script>
+      const changeLang = (languageCode) => {
+       document.documentElement.setAttribute("lang", languageCode);
+      };
+    </script> -->
+    <!-- move this to an external example -->
+
```
Comment (geoffrich) [inline demo/index.html:26]: Do we still need this?

File: src/i18n.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  const observer = new MutationObserver(handleChange);
+  observer.observe(document.documentElement, { attributes: true,
+    attributeFilter: ['lang'] });
+}
+
+/**
+ * Object containing default copy for this element.
+ * Supported languages: en, es.
+ * @private
+ */
```
Comment (geoffrich) [inline src/i18n.js:42]: I don't think this comment is accurate anymore.
```suggestion
```

**Review by geoffrich — COMMENTED** [review]
Just some nitpicks.

**Review by geoffrich — APPROVED** [review]

**Pull request merged**